action/protocol/execution/evm/evm.go

```diff
@@ -639,15 +639,6 @@ func SimulateExecution(
 	)
 
 	ctx = protocol.WithFeatureCtx(ctx)
-	// TODO: move the logic out of SimulateExecution
-	helperCtx := mustGetHelperCtx(ctx)
-	ctx = WithHelperCtx(ctx, HelperContext{
-		GetBlockHash: helperCtx.GetBlockHash,
-		DepositGasFunc: func(context.Context, protocol.StateManager, address.Address, *big.Int, *big.Int) (*action.TransactionLog, error) {
-			return nil, nil
-		},
-		Sgd: nil,
-	})
```

Comment:
After moving this logic out, need to update on all the calling places

Reply:
fixed: add default depositGasFunc when ps.helperCtx.DepositGasFunc not set